asv_bench/benchmarks/index.py

```diff
@@ -63,8 +81,35 @@ def time_as_series_partitions_as_index(
         self.ktk_index.as_flat_series(partitions_as_index=True)
 
 
+class SerializeIndex(IndexBase):
+    timeout = 180
+    params = (
+        [(10 ** 3, 10), (10 ** 4, 100)],  # (values, partitions)
+        [(int, pa.int64())],  # types
+    )
+    param_names = ["number_values__number_partitions", "dtype"]
+
+    def setup(self, number_values__number_partitions, dtype):
+        number_values, number_partitions = number_values__number_partitions
+        return super().setup(number_values, number_partitions, dtype)
+
+    def teardown(self, number_values__number_partitions, dtype):
+        number_values, number_partitions = number_values__number_partitions
+        return super().teardown(number_values, number_partitions, dtype)
+
+    def track_mem_serialized(self, number_values__number_partitions, arrow_type):
+        # Use `sys.getsizeof` as asv's `mem_*` just reports `0` if memory usage is low
+        # enough
+        return sys.getsizeof(pickle.dumps(self.ktk_index))
+
+    def time_serialization(self, number_values__number_partitions, arrow_type):
+        # Time serialization of indices
+        pickle.dumps(self.ktk_index)
```

Comment:
nitpick: I'd actually perform a full roundtrip here. After all, deserialization impacts the performance as well

Reply:
Done. Updated benchmark:
```diff
9,10c9,10
<                           (1000, 10)              (<class 'int'>, DataType(int64))   45.4±0.6ms
<                          (10000, 100)             (<class 'int'>, DataType(int64))   6.16±0.01s
---
>                           (1000, 10)              (<class 'int'>, DataType(int64))   7.04±0.6ms
>                          (10000, 100)             (<class 'int'>, DataType(int64))    454±60ms
```